tests/foreman/api/test_repository.py

```diff
@@ -1206,6 +1208,35 @@ def test_positive_access_protected_repository(self):
             repo_data_file_url, cert=cert_file_path, verify=False)
         self.assertEqual(response.status_code, 200)
 
+    @tier2
+    @upgrade
+    def test_module_stream_repository_crud_operations(self):
+        """Verify that module stream api calls works with product having other type
+        repositories.
+
+        :id: 61a5d24e-d4da-487d-b6ea-9673c05ceb60
+
+        :expectedresults: module stream repo create, update, delete api calls should work with
+        count of module streams
+
+        :CaseImportance: Critical
+        """
+        repository = entities.Repository(
+            url=CUSTOM_MODULE_STREAM_REPO_1,
+            content_type=REPO_TYPE['yum'],
+            product=self.product,
+            unprotected=False,
+        ).create()
+        repository.sync()
+        self.assertGreaterEqual(repository.read().content_counts['module_stream'], 10)
+        repository.url = CUSTOM_MODULE_STREAM_REPO_2
+        repository = repository.update(['name'])
+        repository.sync()
+        self.assertGreaterEqual(repository.read().content_counts['module_stream'], 5)
```

Comment:
Let's assume update or sync silently failed and repository still points to `CUSTOM_MODULE_STREAM_REPO_1`. Since it has at least 10 module streams, this assertion will pass.

Do we control content of `CUSTOM_MODULE_STREAM_REPO_1` and `CUSTOM_MODULE_STREAM_REPO_2`? Can we be sure that one will always have more module streams than the other?

If yes, I would propose to first check number of module streams on repo with smaller number, then update and check with greater number again.

Reply:
yeah ! that is correct. I will update that.